### PR TITLE
Don't allow transforming scene tiles

### LINKED ISSUE #81918: Rotating and flipping tiles doesn't work with Scene Collections.
### Godot version

v4.2.dev5.official [e3e2528ba]

### System information

Godot v4.2.dev5 - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4060 Ti (NVIDIA; 31.0.15.3713) - 11th Gen Intel(R) Core(TM) i5-11400F @ 2.60GHz (12 Threads)

### Issue description

The newly re-introduced per-tile rotation and flipping does not work with scene collections, making a different colored error box depending on the state of the flip. In this case the scene is a simple sprite with a collision shape.
image

### Steps to reproduce

Make a Scenes Collection in a Tileset, and try flipping or/and rotating one of the tiles.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #81918
image

### CODE REVIEW

File: editor/plugins/tiles/tile_map_editor.cpp [hunk 1]
```diff
@@ -107,17 +107,25 @@ void TileMapEditorTilesPlugin::_update_transform_buttons() {
 		return;
 	}
 
-	if (tile_set->get_tile_shape() == TileSet::TILE_SHAPE_SQUARE || selection_pattern->get_size() == Vector2i(1, 1)) {
-		transform_button_rotate_left->set_disabled(false);
-		transform_button_rotate_left->set_tooltip_text("");
-		transform_button_rotate_right->set_disabled(false);
-		transform_button_rotate_right->set_tooltip_text("");
+	if (!selection_pattern->is_empty() && Object::cast_to<TileSetScenesCollectionSource>(tile_set->get_source(selection_pattern->get_cell_source_id(Vector2i())).ptr())) {
```
Comment (groud) [inline editor/plugins/tiles/tile_map_editor.cpp:110]: You need test all cells of the pattern to see if at least one is a scene, not simply the one in the `(0, 0)` position.

**Review by groud — CHANGES_REQUESTED** [review]

File: editor/plugins/tiles/tile_map_editor.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/plugins/tiles/tile_map_editor.cpp:110]: For some reason I assumed you can't mix them .-.

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by ettiSurreal** [comment]
Just to make sure, is this temporary, or is the ability to transform scene tiles not planned whatsoever?

**Comment by KoBeWi** [comment]
It was discussed here https://chat.godotengine.org/channel/editor?msg=zKCnZFgvEBovXWoJd (you need to wait a bit for the message to load)
For now there are no plans, as it's not that straightforward. Only rotation then be added somewhat reliably, flipping could be achieved with scale, but it shouldn't be used for physics objects. Another thing is that reserving bit for rotation will limit the number of scenes you can put in a collection (they rely only on alternative IDs).

**Comment by groud** [comment]
> Just to make sure, is this temporary, or is the ability to transform scene tiles not planned whatsoever?

It could be done, but a bit technical to implement (and has some annoying drawbacks, like the impossibility to animate the scene position). So for now the simple solution is to disable it, even if it could maybe be implemented later if there's enough demand for it.

**Pull request merged**